File: msgraph-mail/typescript/src/users/internetMessageHeader.ts
import {SerializationWriter, ParseNode, Parsable} from '@microsoft/kiota-abstractions';

export class InternetMessageHeader implements Parsable {
    /** Stores additional data not described in the OpenAPI description found when deserializing. Can be used for serialization as well.  */
    public readonly additionalData: Map<string, unknown> = new Map<string, unknown>();
    /** Represents the key in a key-value pair.  */
    public name?: string | undefined;
    /** The value in a key-value pair.  */
    public value?: string | undefined;
    /**
     * The deserialization information for the current model
     * @returns a Map<string, (item: T, node: ParseNode) => void>
     */
    public getFieldDeserializers<T> () : Map<string, (item: T, node: ParseNode) => void> {
        return new Map<string, (item: T, node: ParseNode) => void>([
            ["name", (o, n) => { (o as unknown as InternetMessageHeader).name = n.getStringValue(); }],
            ["value", (o, n) => { (o as unknown as InternetMessageHeader).value = n.getStringValue(); }],
        ]);
    };
    /**
     * Serialiazes information the current object
     * @param writer Serialization writer to use to serialize this model
     * @returns a void
     */
    public serialize (writer: SerializationWriter) : void {
        writer.writeStringValue("name", this.name);
        writer.writeStringValue("value", this.value);
        writer.writeAdditionalData(this.additionalData);
    };
}

File: msgraph-mail/dotnet/Users/InternetMessageHeader.cs
using Kiota.Abstractions.Serialization;
using System;
using System.Collections.Generic;
using System.Linq;
namespace Graphdotnetv4.Users {
    public class InternetMessageHeader : IParsable<InternetMessageHeader> {
        /// <summary>Stores additional data not described in the OpenAPI description found when deserializing. Can be used for serialization as well.</summary>
        public IDictionary<string, object> AdditionalData { get; private set; } = new Dictionary<string, object>();
        /// <summary>The serialization information for the current model</summary>
        public IDictionary<string, Action<InternetMessageHeader, IParseNode>> DeserializeFields => new Dictionary<string, Action<InternetMessageHeader, IParseNode>> {
            {
                "name", (o,n) => { o.Name = n.GetStringValue(); }
            },
            {
                "value", (o,n) => { o.Value = n.GetStringValue(); }
            },
        };
        /// <summary>Represents the key in a key-value pair.</summary>
        public string Name { get; set; }
        /// <summary>The value in a key-value pair.</summary>
        public string Value { get; set; }
        /// <summary>
        /// Serialiazes information the current object
        /// <param name="writer">Serialization writer to use to serialize this model</param>
        /// </summary>
        public void Serialize(ISerializationWriter writer) {
            writer.WriteStringValue("name", Name);
            writer.WriteStringValue("value", Value);
            writer.WriteAdditionalData(AdditionalData);
        }
    }
}
